ibm/resource_ibm_network_gateway.go

```diff
@@ -0,0 +1,896 @@
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"notes": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"post_install_script_uri": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				Default:          nil,
+				ForceNew:         true,
+				DiffSuppressFunc: applyOnce,
+			},
+
+			"tags": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+
+			"os_reference_code": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "OS_VYATTA_5600_5_X_UP_TO_1GBPS_SUBSCRIPTION_EDITION_64_BIT",
+				ForceNew: true,
+			},
+
+			"datacenter": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+			},
+
+			"network_speed": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  100,
+				ForceNew: true,
+			},
+			//Does not apply for Network gateway but is required for cancellation
+			"hourly_billing": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+				ForceNew: true,
+			},
+
+			"tcp_monitoring": {
+				Type:             schema.TypeBool,
+				Optional:         true,
+				Default:          false,
+				ForceNew:         true,
+				DiffSuppressFunc: applyOnce,
+			},
+
+			"redundant_power_supply": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			// Monthly only
+			"package_key_name": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				ForceNew:         true,
+				Default:          "NETWORK_GATEWAY_APPLIANCE",
+				DiffSuppressFunc: applyOnce,
+			},
+
+			// Monthly only
+			"process_key_name": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				ForceNew:         true,
+				Default:          "INTEL_SINGLE_XEON_1270_3_40_2",
+				DiffSuppressFunc: applyOnce,
+			},
+
+			// Monthly only
+			"os_key_name": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				ForceNew:         true,
+				Default:          "OS_VYATTA_5600_5_X_UP_TO_1GBPS_SUBSCRIPTION_EDITION_64_BIT",
+				DiffSuppressFunc: applyOnce,
+			},
+
+			// Monthly only
+			"disk_key_names": {
+				Type:             schema.TypeList,
+				Optional:         true,
+				ForceNew:         true,
+				Elem:             &schema.Schema{Type: schema.TypeString},
+				DiffSuppressFunc: applyOnce,
+			},
+
+			// Monthly only
+			"redundant_network": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+				ForceNew: true,
+			},
+
+			// Monthly only
+			"unbonded_network": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+				ForceNew: true,
+			},
+
+			// Monthly only
+			"public_bandwidth": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				ForceNew:         true,
+				Default:          "BANDWIDTH_20000_GB",
```

Comment:
Won't it make more sense to have it just 20000 and you compose the actual value in the code?
Also are we sure that this default works in all the cases? You can refer to `public_bandwidth` attribute usage in `ibm_bare_metal` resource.